runtime/src/plugin.rs

```diff
@@ -55,6 +55,43 @@ fn profiling_strategy() -> ProfilingStrategy {
     }
 }
 
+fn calculate_available_memory(
+    available_pages: &mut Option<u32>,
+    modules: &BTreeMap<String, Module>,
+) -> anyhow::Result<()> {
+    let available_pages = match available_pages {
+        Some(p) => p,
+        None => return Ok(()),
+    };
+
+    for (name, module) in modules.iter() {
+        let mut memories = 0;
+        for export in module.exports() {
+            if let Some(memory) = export.ty().memory() {
+                let memory_max = memory.maximum();
+                match memory_max {
+                    None => anyhow::bail!("Unbounded memory in module {name}, when `memory.max_pages` is set all modules \
+                                           must have a maximum bound set on an exported memory"),
+                    Some(m) => {
+                        *available_pages = available_pages.saturating_sub(m as u32);
+                        if *available_pages == 0 {
+                            anyhow::bail!("Not enough memory configured to run the provided plugin");
```

Comment:
Could we perhaps provide some more hints in these error messages? Maybe let the user know they should look at the manifest max_pages config?